### ISSUE TITLE
Override existing rule severity

### ISSUE THREAD

**Issue by Urmil-Ally** [detekt/detekt#3129]
I would like to know if there is a simpler way to override the severity of existing rules in detekt. 
I am aware of writing custom rules but I am looking for an easy way to do so via yml file where one can activate or deactivate rules with one line. 

For example: 
```
style
  ClassOrdering:
```
This is reported as `Style` severity but I would like to override it to `Warning` or `error` for it to work with Danger.js

Any help is appreciated.

**Comment by schalkms** [comment]
Yep, there's a way to do this. Detekt supports build failure thresholds for this exact reason.
Please take a look at the documentation on the website.
https://detekt.github.io/detekt/failonbuild.html

Please let us kindly know whether this helps you.
Otherwise there's already an open issue for supporting different rule severities (#533).
If this is what you are trying to achieve, feel free to vote for the mentioned feature.

**Comment by Urmil-Ally** [comment]
Hey @schalkms. Thank you for the response. But this (fail builds on thresholds) is not what I am looking for. 

The ask is to be able to change Severity levels for a particular rule/ruleset so that Danger can fail the build if that rule is violated. 

So, if I use the above, if a developer violates `Class Ordering`, Detekt reports it as `Style` which Danger picks it up as `Warning` and allows the build to proceed. Similarly, there are few rules that Detekt reports as `Defect` but it suites better as `Warning` for our codebase. 

#533 talks about it but it seems it hasn't been worked upon yet.

**Comment by Urmil-Ally** [comment]
Well, I am not so sure if it's a Danger or Danger-detekt plugin thing. But this is what has been observed so far: 
Danger does not honor all sev levels in the [library](https://github.com/NFesquet/danger-kotlin_detekt). 
It just identifies Warning, Error, and Fatal. 

Keeping danger aside, in general, what would be a better way to override severity to be able to reduce it merely two - `Warning` and `Error`? So as to fail builds only on `Error`.

**Comment by schalkms** [comment]
Okay, good to know. Thanks for providing insights. Danger is an excellent 3rd party plugin. However, because it’s a 3rd plugin, the detekt team doesn’t officially support it. 
Why do you not raise an issue in the danger repository asking for support?
May I also please ask you to add your suggestion to the linked issue in order to have the conversation in one place?
